### PR TITLE
Skip disabled controls when moving to first/last item

### PR DESCRIPTION
**Pull request by pr8x**
## What does the pull request do?

Skip disabled/non-focusable controls when navigating to first/last item. This is required to make right arrow navigation in menus work when the first subitem is disabled. 

UWP:

uwp_flyout_right_nav

## What is the current behavior?

Pressing Right arrow on menu item simply does nothing when the first subitem is disabled.

## What is the updated/expected behavior with this PR?

Pressing Right Arrow on a menu item should focus the next available item of the submenu.

## Checklist

- [X] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

### CODE REVIEW

File: src/Avalonia.Controls/ItemsControl.cs [hunk 1]
```diff
@@ -502,25 +502,34 @@ private void UpdatePseudoClasses(int itemCount)
...
-            } while (c != null && c != from && direction != NavigationDirection.First && direction != NavigationDirection.Last);
+                direction = direction switch
+                {
+                    //We did not find an enabled first item. Move downwards until we find one.
+                    NavigationDirection.First => NavigationDirection.Down,
+
+                    //We did not find an enabled last item. Move upwards until we find one.
+                    NavigationDirection.Last => NavigationDirection.Up,
+                    _ => direction
+                };
```
Comment (maxkatz6) [inline src/Avalonia.Controls/ItemsControl.cs:529]: What will happen if all items are disabled, won't it be trapped in the cycle? Minimal test might be useful.

Comment (pr8x) [reply src/Avalonia.Controls/ItemsControl.cs]: Good spot. This was actually broken. I disabled wrapping in that case now so it should terminate correctly.

Comment (pr8x) [reply src/Avalonia.Controls/ItemsControl.cs]: Also added some tests.

**Comment by pr8x** [comment]
There's one more broken interaction: When all subitems are disabled I cannot close the submenu with Arrow-Left again.

**Review by maxkatz6 — APPROVED** [review]

**Comment by maxkatz6** [comment]
@pr8x seems like related tests still fail

**Comment by pr8x** [comment]
@maxkatz6 Fixed UTs.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0022287-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0022299-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Pull request merged**